test/src/main/java/org/apache/accumulo/test/functional/WALSunnyDayIT.java

```diff
@@ -232,4 +216,53 @@ private void writeSomeData(Connector conn, String tableName, int row, int col) t
     return result;
   }
 
+  private final int TIMES_TO_COUNT = 20;
+  private final int PAUSE_BETWEEN_COUNTS = 100;
+
+  private Map<String,Boolean> getWALsAndAssertCount(Connector c, int expectedCount) throws Exception {
+    // see https://issues.apache.org/jira/browse/ACCUMULO-4110. Sometimes this test counts the logs before
+    // the new standby log is actually ready. So let's try a few times before failing, returning the last
+    // wals variable with the the correct count.
+    Map<String,Boolean> wals = _getWals(c);
+    if (wals.size() == expectedCount) {
+      return wals;
+    }
+
+    int waitLonger = getWaitFactor();
+    for (int i = 1; i <= TIMES_TO_COUNT; i++) {
+      Thread.sleep(i * PAUSE_BETWEEN_COUNTS * waitLonger);
+      wals = _getWals(c);
+      if (wals.size() == expectedCount) {
+        return wals;
+      }
+    }
+
+    fail("Unable to get the correct number of WALs, expected " + expectedCount + " but got " + wals.toString());
```

Comment:
Can you clarify what you are asking here @joshelser?  I replaced existing asserts like the following

assertEquals(wals.toString(), 2, wals.size());

with this new method named getWALsAndAssertCount.  I still think it is an assert, just an assert that tries extra hard to pass.